polkadot/node/network/collator-protocol/src/validator_side_experimental/mod.rs

```diff
@@ -148,17 +157,20 @@ async fn run_inner<Context>(mut ctx: Context, mut state: State<Db>) -> FatalResu
 			},
 			resp = state.collation_response_stream().select_next_some() => {
 				state.handle_fetched_collation(ctx.sender(), resp).await;
-			}
+			},
+			_ = &mut timer => {
+			},
 		}
 
 		// Now try triggering advertisement fetching, if we have room in any of the active leaves
 		// (any of them are in Waiting state).
 		// We could optimise to not always re-run this code (have the other functions return
-		// whether or not we should attempt launching fetch requests) However, most messages could
+		// whether we should attempt launching fetch requests) However, most messages could
 		// indeed trigger a new legitimate request.
 		// Also, it takes constant time to run because we only try launching new requests for
 		// unfulfilled claims. It's probably not worth optimising.
-		state.try_launch_new_fetch_requests(ctx.sender()).await;
+		let maybe_delay = state.try_launch_new_fetch_requests(ctx.sender()).await;
+		timer = create_timer(maybe_delay.map(|delay| std::cmp::max(delay, 500)));
```

Comment:
Why wake up early if we know for sure that no advertisement will be ready until `maybe_delay`?

Reply:
Sorry, my bad. In my head I interpreted it the other way around 🙈 

That's exactly how it should work. Wake up after delay but no earlier than 500ms.